skcosmo/SparseMethods/Sparsified.py

```diff
@@ -0,0 +1,108 @@
+    """
+    Super-class defined sparcified methods
+
+        :param kernel: the kernel used for this learning
+        :param gamma: exponential factor of the rbf and sigmoid kernel
+        :param degree: polynomial kernel degree
+        :param coef0: free term of the polynomial and sigmoid kernel
+        :param kernel_params: kernel parameter set
+        :param n_active: the size of the small dataset used in learning
+        :param center: if True, centering of kernel during the learning is carried out
+        :param n_jobs: The number of jobs to use for the computation the kernel. This works by breaking down the pairwise matrix into n_jobs even slices and computing them in parallel.
+        :param selector: defines the sampling method for the active subsets
+        :param selector_args: define the parameters for selector function
+    """
+
+    def __init__(
+        self,
+        kernel="linear",
+        gamma=None,
+        degree=3,
+        coef0=1,
+        kernel_params=None,
+        n_active=None,
+        center=True,
+        n_jobs=1,
+        selector=SampleFPS,
+        selector_args={},
+    ):
+        """
+        Initializes superclass for sparse methods
+
+        :param kernel: the kernel used for this learning
+        :param gamma: exponential factor of the rbf and sigmoid kernel
+        :param degree: polynomial kernel degree
+        :param coef0: free term of the polynomial and sigmoid kernel
+        :param kernel_params: kernel parameter set
+        :param n_active: the size of the active dataset used in learning
+        :param center: if True, centering of kernel during the learning is carried out
+        :param n_jobs: The number of jobs to use for the computation the kernel. This works by breaking down the pairwise matrix into n_jobs even slices and computing them in parallel.
+        :param selector: defines the sampling method for the active subsets
+        :param selector_args: define the parameters for selector function
+        """
+        self.kernel = kernel
+        self.gamma = gamma
+        self.degree = degree
+        self.coef0 = coef0
+        self.kernel_params = kernel_params
+        self.n_active = n_active
+        self.center = center
+        self.n_jobs = n_jobs
+        self._selector = selector
+        self._selector_args = selector_args
+
+    def _get_kernel(self, X, Y=None):
+        """
+        Calculate kernel for the matrix X or (optionally) for matrix X and Y
+        :param X: matrix, for which we calculate kernel
+        :param Y: A second feature array only if X has shape [n_samples_a, n_features].
+
+        :return: sklearn.metrics.pairwise.pairwise_kernels(X, Y)
+        """
+        if self.kernel == "precomputed":
+            if X.shape[-1] != self.n_active:
+                raise ValueError("The supplied kernel does not match n_active.")
+            return X
+        if callable(self.kernel):
+            params = self.kernel_params or {}
```

Comment:
I made the default for `kernel_params = {}`, so we no longer need the `or {}`
```suggestion
            params = self.kernel_params
```